src/main/java/ca/team2706/frc/robot/commands/drivebase/DriverAssistVision.java

```diff
@@ -14,189 +15,335 @@
      * Network table instance to get data from vision subsystem
      */
     private NetworkTableInstance inst;
 
     /**
-     * Network table to get data from vision subsystem
+     * Network table to get Pathfinder data from vision subsystem
      */
-    private NetworkTable table;
+    private NetworkTable pathfinderTable;
 
     /**
-     * Trajectory generated to move robot to target
+     * True if ring light on stage of execute() method has completed, false otherwise
      */
-    private Trajectory traj;
+    private boolean ringLightOnStageComplete = false;
 
     /**
-     * True if trajectory has been generated for this command, false otherwise.
-     * (This will be removed when motion control functionality is implemented.)
+     * Current value of time delay after ring light has been turned on.
+     */
+    private double ringLightOnDelayTime = 0.0;
+
+    /*
+     * Network table entry named TapeDetected, a boolean set to true if the tape target has been
+     * detected by the vision system, false otherwise
+     */
+    private NetworkTableEntry tapeDetectedEntry;
+
+    /*
+     * Type of target that robot will move to (CARGO_AND_LOADING, ROCKET, or BALL)
      */
-    private boolean trajGenerated = false;
+    private DriverAssistVisionTarget target;
 
     /**
-     * True if vision system has been assessed to be offline
+     * Trajectory generated by Pathfinder to move robot to target. Declared volatile since it is shared between
+     * different threads.
      */
-    boolean visionOffline = false;
+    private volatile Trajectory trajectory;
 
     /**
      * Creates empty driver assist command object (needed for unit testing framework only)
      */
     public DriverAssistVision() {
         // Ensure that this command is the only one to run on the drive base
-        requires(DriveBase.getInstance());
+        requires(RingLight.getInstance());
 
         setupNetworkTables();
 
         commandAborted = false;
-        trajGenerated = false;
-        visionOffline = false;
-        angYawTargetWrtCameraLOSCWposPrev = 0.0;
-        distanceCameraToTarget_CameraPrev = 0.0;
+        generateTrajectoryRequestStageComplete = false;
+        ringLightOnDelayTime = 0.0;
+        ringLightOnStageComplete = false;
     }
 
     /**
      * Creates the driver assist command
      *
-     * @param driverAssistCargoAndLoading: True if target is cargo ship or loading bay, false otherwise
-     * @param driverAssistRocket:          True if target is a rocket ship, false otherwise
+     * @param DriverAssistVisionTarget target: The type of the destination target (CARGO_AND_LOADING, ROCKET, BALL)
      */
-    public DriverAssistVision(boolean driverAssistCargoAndLoading, boolean driverAssistRocket) {
-        this.driverAssistCargoAndLoading = driverAssistCargoAndLoading;
-        this.driverAssistRocket = driverAssistRocket;
+    public DriverAssistVision(DriverAssistVisionTarget target) {
+        this.target = target;
```

Comment:
It might be easier just to say
```Java
this();
this.target = target;
```

Reply:
Sorry, @eandr127, do you mean put
`this();`

in front of

`this.target = target`?

What does `this()` do?

Reply:
Yes. Since your code in this constructor is identical to the other constructor, you can use `this()` to call the other constructor. That way you can get rid of the rest of the code in this constructor.

Reply:
Updated.